src/domain/common/port/RequestExecutorPort.ts

```diff
@@ -0,0 +1,7 @@
+import Request from '@core/domain/request/Request';
+import Response from '@core/domain/response/Response';
+
+export default interface RequestExecutorPort {
+    execute(request: Request, response: Response): Promise<Response>;
```

Comment:
Why would we both accept and return a Response? Is response an in/out param?

Reply:
It was to conform to existing error handling, but after your comment I got idea on how to design it so that I can remove `request` param.